packages/react-core/src/components/TextInputGroup/TextInputGroupMain.tsx

```diff
@@ -9,8 +10,25 @@ export interface TextInputGroupMainProps extends React.HTMLProps<HTMLDivElement>
   className?: string;
   /** Icon to be shown on the left side of the text input group main container */
   icon?: React.ReactNode;
+  /** Type that the input accepts. */
+  type?:
+    | 'text'
+    | 'date'
+    | 'datetime-local'
+    | 'email'
+    | 'month'
+    | 'number'
+    | 'password'
+    | 'search'
+    | 'tel'
+    | 'time'
+    | 'url';
   /** Callback for when there is a change in the input field*/
   onChange?: () => void;
+  /** Callback for when the input field is focused*/
+  onFocus?: () => void;
```

Comment:
On the TextInput component, we add the event as a parameter for the event handlers so the consumer can use it if the want. I think that might be wise here for onChange, onFocus, and onBlur. In fact, onChange might benefit from also passing back the new value of the input via a parameter like the TextInput component does.

Reply:
I can update the types to all include events, but adding the input value as a parameter itself will require adding a handler to TextInputGroupMain as was done in TextInput because otherwise typescript will complain about `onChange` not being assignable to the `onChange` on `input`.

I'm happy to have onChange just take `event` as a parameter, or add `event` and the `value` as paramaters and add a handler if you think that's best. I would lean towards just having `event` and letting the consumer get the value out of it if they choose though.

Reply:
I think keeping the types consistent with `TextInput` actually does help with upgrading and migrating to the TextInputGroup for products in the future.

`TextInput` handles the issue you mention by defining [this function](https://github.com/patternfly/patternfly-react/blob/320fde1581d401ba386289ac2b6893d567a079b8/packages/react-core/src/components/TextInput/TextInput.tsx#L106)